lib/Sema/TypeCheckType.cpp

```diff
@@ -988,9 +999,10 @@ static Type resolveTypeDecl(TypeDecl *typeDecl, SourceLoc loc,
 
   // Resolve the type declaration to a specific type. How this occurs
   // depends on the current context and where the type was found.
-  Type type =
-    TypeChecker::resolveTypeInContext(typeDecl, foundDC, resolution, options,
-                                      generic);
+  if (!type)
+    type =
+      TypeChecker::resolveTypeInContext(typeDecl, foundDC, resolution,
```

Comment:
Should getting the structural type be pushed down into resolveTypeInContext(), since it has access to the `resolution` object?

Reply:
We probably could, but we also need to avoid the call to the previous `resolveDeclSignature` because it triggers `validateDecl`.